addons/survey/static/src/js/survey_form.js

```diff
@@ -1109,6 +1112,20 @@ publicWidget.registry.SurveyFormWidget = publicWidget.Widget.extend(SurveyPreloa
                 $(dependingQuestion).find('textarea').val('');
             }
         }
+        // One page sections visibility
+        if (this.options.questionsLayout !== 'one_page' || !this.options.pageIds) {
+            return;
+        }
+        for (const sectionId of this.options.pageIds) {
+            const sectionInfo = document.querySelector(`.js_section_wrapper[id="${sectionId}"]`);
+            const sectionQuestions = document.querySelectorAll(`.js_question-wrapper[page-id="${sectionId}"]`);
+            const sectionHasVisibleQuestions = Array.from(sectionQuestions).some(question => !question.classList.contains('d-none'));
```

Comment:
You could use `all` instead of `some` and remove the `!` inside the callback to do less operations.